WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanViewModel.kt

```diff
@@ -1,15 +1,65 @@
 package org.wordpress.android.ui.jetpack.scan
 
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModel
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.scan.ScanStateModel
+import org.wordpress.android.fluxc.model.scan.ScanStateModel.State
+import org.wordpress.android.ui.jetpack.scan.ScanListItemState.ScanState.ScanIdleState
+import org.wordpress.android.ui.jetpack.scan.ScanListItemState.ScanState.ScanScanningState
+import org.wordpress.android.ui.jetpack.scan.ScanViewModel.UiState.Content
 import javax.inject.Inject
 
-class ScanViewModel @Inject constructor() : ViewModel() {
+class ScanViewModel @Inject constructor(
+    private val scanStatusService: ScanStatusService
+) : ViewModel() {
     private var isStarted = false
+    private var lastScanState: State? = null
```

Comment:
Idea (💡 ): It initially confused me a bit, thinking that the `State` here is the `UiState`. I am still not used to that I guess. Since those two are really close, readability-wise, what if the full qualifier is used instead, like `ScanStateModel.State`?

PS: The same applies everywhere within this view model. I understand that this is more verbose and could make the `when (model.state) { ... }` look funnier, but I have a suggestion there as well. 😊

Reply:
Nice suggestion 👍 - full qualifier added in 0149797